docs/guides/tgi.mdx

```diff
@@ -0,0 +1,140 @@
+$ truss init ./my-tgi-trus --backend TGI
+? model_id
+```
+
+One thing that makes Truss support in TGI a bit different from the default is that there isn't any model instantiation or model inference code that needs to be written! Because we leverage TGI under the hood, TGI handles loading the model in and running inference. All you need to do is pass in the model ID as it appears on Huggingface.
+
+After typing in your model ID, you'll be prompted with the `endpoint` you'd like to use.
+```
+$ truss init ./my-tgi-trus --backend TGI
+? model_id facebook/opt-125M
+? endpoint
+```
+
+TGI supports 2 endpoints:
+- `generate` which returns the entire generated response upon completion
+- `generate_stream` which streams the response as it's being generated
+
+
+You can press the `tab` key on any of these dialogues to see options for values.
+
+
+Finally, you'll be asked for the name of your model.
+
+# Deploying your TGI model
+
+Now that we have a TGI model, let's deploy this model to Baseten and see how it performs.
+
+You'll need an API key to deploy your model. You can get one by navigating to your Baseten settings [page](https://app.baseten.co/settings/account/api_keys). To push the model to Baseten, run the following command:
+```
+(truss-py3.9) @aspctu $ truss push --publish
+```
+
+Let's walk through what this command is doing
+```
+- "truss push": This CLI command zips and uploads your Truss to Baseten
+- "--publish": This tells Baseten that you want a Production endpoint (vs. a development environment)
+```
+
+After running this command, you'll be prompted to pass in your API key. Once you've done that, you can view your deployment status from the Baseten dashboard. Once the model has been deployed, you can invoke it by running the following command:
+
+```
+truss predict -d '{"inputs": "What is a large language model?", "parameters": {"max_new_tokens": 128, "sample": true}} --published'
+```
+Let's walk through what this command is doing
+```
+- "truss predict": This CLI command invokes your model
+- "-d": This tells the CLI that the next argument is the data you want to pass to your model
+- '{"inputs": "What is a large language model?", "parameters": {"max_new_tokens": 128, "sample": true}}': This is the data you want to pass to your model. TGI expects a JSON object with 2 keys: `inputs` and `parameters` (optional). `inputs` is the prompt you want to pass to your model and `parameters` is a JSON object of parameters you want to pass to your model.
+- "--published": This tells the CLI that you want to invoke the production endpoint (vs. a development endpoint)
+```
+
+# Tuning your TGI server
+
+After deploying your model, you may notice that you're not getting the performance you'd like out of TGI. If you navigate to the `target_directory` from above, you'll find a `config.yaml` file that contains a key `build`. The following is a set of arguments you can pass to the `build` key to tune TGI for max performance:
+
+1. `max_input_length`
```

Comment:
couple things, for each of these:
* include in this doc what the default is if there is one
* Phrase each of these as a "trade-off". ie: higher values of X mean this benefit and this drawback, while lower values mean this other benefit and this other drawback